core/commands/storage/upload/upload/do_waitupload.go

```diff
@@ -83,11 +84,17 @@ func waitUpload(rss *sessions.RenterSession, offlineSigning bool, renterId strin
 					return err
 				}
 				num := 0
+				m := make(map[string]int)
 				for _, c := range meta.Contracts {
+					m[c.State.String()]++
```

Comment:
maybe better not adding an overall info, but each shard struct can have an additional contract state field